packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts

```diff
@@ -732,6 +732,13 @@ export interface ServiceProps {
    */
   readonly vpcConnector?: IVpcConnector;
 
+  /**
+   * Specifies whether your App Runner service is publicly accessible.
+   *
+   * @default true
+   */
+  readonly isPubliclyAccessible?: boolean;
```

Comment:
I'm not too familiar with restrictions for creating a VPC ingress connection but is it still possible to create an ingress connection if `isPubliclyAccessible` is set to `true`? Just trying to figure out why this property is needed, and if that is the case that it must be set to false to create an ingress connection, we should check that somewhere when creating the `VpcIngressConnection` class constructor.

Reply:
To use VPC Ingress Connection, the Service's `isPubliclyAccessible` needs to be set to false.
However, since the flow is to create the VPC Ingress Connection after creating the Service, we can only set it to false in the Service configuration.

I've added this restriction to README and property docs.

I considered checking the Service's i`isPubliclyAccessible` in the `VpcIngressConnection`, but I thought it wasn't possible to retrieve this since it doesn't exist as a [CFn attribute](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apprunner.CfnService.html#properties).
Is there any good way to check this?

Reply:
Hmmm ya I can't really think of a good way to check it either. Will CloudFormation throw an error if you try to create a `VpcIngressConnection` with `isPubliclyAcessible` set to true? If so then I think just documenting the restriction is the best we can do on the CDK side.

Reply:
@paulhcsun 
Thanks.
Yeas, I tried to create `VpcIngressConnection` with `isPubliclyAcessible` set to `true` and got the following an error:

> Resource handler returned message: "Failed to create a VpcIngressConnection for a Service because it is publicly accessible.

Reply:
Ok in that case I think it is okay to leave it as is then. Thanks for checking!